src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const convertedObject = {};
+
+  const splited = sourceString.split(';');
+
+  splited.forEach((split) => {
+    if (split.trim()) {
+      const [property, value] = split.split(':');
+
+      convertedObject[property.trim()] = value.trim();
+    }
+  });
```

Comment:
Instead of using 'forEach' and an external object, you can use 'reduce' to directly transform the array into the desired object. This would adhere to the checklist point about avoiding the modification of an existing array/object and using iteration methods that return a new array/object.